src/PSDMatrices.jl

```diff
@@ -1,13 +1,64 @@
+
 module PSDMatrices
 
-using LinearAlgebra
-import LinearAlgebra: cholesky, diag, det, logdet
+import Base: \, /, size, inv, copy, ==
+import LinearAlgebra: cholesky, det, logdet, qr, triu, diag, UpperTriangular
+
+struct PSDMatrix{T,Ltype} <: AbstractMatrix{T}
+    L::Ltype
+    PSDMatrix(L::AbstractMatrix{T}) where {T} = new{T,typeof(L)}(L)
+end
+
+# Custom functions
+
+function todense(M::PSDMatrix)
+    return M.L * M.L'
+end
+
+
+function X_A_Xt(A::PSDMatrix, X::AbstractMatrix)
+    return PSDMatrix(X * A.L)
+end
+
+
+function choleskify_factor(M::PSDMatrix)
```

Comment:
I hope that the "choleskify" (instead of a cholesky()) makes it clear that there is no promise of  [O(n^3/3) complexity](https://en.wikipedia.org/wiki/Cholesky_decomposition#Computation). But the naming is certainly up for debate :)

Reply:
I like the name :)